### DISCUSSION TITLE
How to remove Remove unused JS & CSS

### DISCUSSION THREAD

**Discussion by bhatvikrant** [vercel/next.js#21262] [Help]
If you go to [this website](https://coox-website-beta-next.vercel.app/) and run a lighthouse test, under the performance report you can see the following suggestions for improvement:

![Screenshot 2021-01-18 at 6 33 15 AM](https://user-images.githubusercontent.com/50735025/104861740-0cff0b00-5957-11eb-9f67-6efdee5bad77.png)

---

**The 2 main areas of concern for me are:**

---

- [ ] **1. Remove unused JavaScript**

In the screenshot below, it suggests me to use `React.lazy()` for code splitting, but nextjs already does route based code-splitting. *Does this mean I need to lazy load the components that I am using in every specific page?*

![Screenshot 2021-01-18 at 6 34 34 AM](https://user-images.githubusercontent.com/50735025/104861791-3d46a980-5957-11eb-9c43-737d2d43695d.png)

---

- [ ] **2. Remove unused CSS**

_I understand that I need to purge the unused CSS, but how can I do that in nextjs?_ 
In this project, I am using bootstrap so solutions specific to that would be really helpful

![Screenshot 2021-01-18 at 6 41 03 AM](https://user-images.githubusercontent.com/50735025/104862050-248ac380-5958-11eb-89ce-6bdc13ece581.png)

**Comment by fluentmoheshwar** [comment]
1. Remove unused JavaScript
 You don't need to do it. It says if you are not using server-side rendering.
 2. Remove unused CSS
 There are many packages in npm to do that.

**Comment by GabenGar** [comment]
The CSS purging tools come with the caveat that they will purge any declaration not present in the final build files, aka any string concatenation from arrays/string constructor functions is going to be assumed as unused by them.
